frontend/src/modules/dataset/dataset-browser.tsx

```diff
@@ -0,0 +1,389 @@
+            setState({
+                type: DatasetActionType.Pagination,
+                payload: { currentPageIndex: state.pagination.pagesCount }
+            });
+        }
+    }, [state.pagination]);
+
+    // if items or filteringText changes recompute filteredItems
+    useEffect(() => {
+        setState({
+            type: DatasetActionType.Filter,
+            payload: {
+                filteringText: state.filter.filteringText,
+                filteringTextDisplay: state.filter.filteringTextDisplay,
+                filteredItems: state.items.filter((item) => {
+                    return !!item.name?.includes(state.filter.filteringText);
+                })
+            }
+        });
+    }, [state.items, state.filter.filteringText, state.filter.filteringTextDisplay]);
+
+    useEffect(() => {
+        onItemsChange?.(new CustomEvent('itemsChange', {detail: {items: state.items}}));
+    }, [state.items, onItemsChange]);
+
+    // create the appropriate table properties for the current viewing context (scopes, listing datasets, listing a dataset resources)
+    const displayModeTableProperties = tablePropertiesForDisplayMode(displayMode, state, setState);
+
+    return (
+        <Box>
+            <SpaceBetween direction='vertical' size='s'>
+                <Condition condition={header !== undefined}>
+                    { header }
+                </Condition>
+                
+                <Condition condition={state.manageMode !== DatasetBrowserManageMode.Create}>
+                    <BreadcrumbGroup items={state.breadcrumbs} onClick={(event) => {
+                        event.preventDefault();
+                        
+                        setState({
+                            type: DatasetActionType.DatasetContext,
+                            payload: event.detail.href.length > 0 ? JSON.parse(event.detail.href) : undefined
+                        });
+                    }} />
+                </Condition>
+
+                <Table
+                    // add common properties shared by every display mode
+                    header={<Header actions={actions?.({
+                        items: state.items,
+                        manageMode: state.manageMode,
+                        selectedItems: state.selectedItems,
+                        datasetContext: state.datasetContext,
+                        filter: state.filter,
+                    }, setState)} counter={
+                        state.selectedItems?.length > 0
+                            ? `(${state.selectedItems.length}/${state.items.length})`
+                            : `(${state.items.length})`
+                    }>{displayMode}s</Header>}
+                    empty={state.manageMode === DatasetBrowserManageMode.Create ? EmptyState('No files uploaded') : EmptyState('No Entries exist')}
+                    filter={
+                        <MLSTextFilter
```

Comment:
Why is this necessary? Why does the default filter not suffice?

Reply:
Unfortunately the default doesn't give you access to keyboard events, it only gives you access to text changes. So using the existing TextFilter to provide parity with the AWS S3 Console is impossible.

Reply:
I think we covered this in our dev sync about the limitations of the original.